lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Migration.hs

```diff
@@ -0,0 +1,143 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE RankNTypes #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: Apache-2.0
+--
+-- This module contains an algorithm to select coins for migration from legacy
+-- wallets to newer wallets.
+
+module Cardano.Wallet.Primitive.CoinSelection.Migration
+    ( selectCoinsForMigration
+    , idealBatchSize
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.CoinSelection
+    ( CoinSelection (..)
+    , CoinSelectionOptions (..)
+    , changeBalance
+    , inputBalance
+    )
+import Cardano.Wallet.Primitive.Fee
+    ( Fee (..), FeeOptions (..) )
+import Cardano.Wallet.Primitive.Types
+    ( Coin (..), TxIn (..), TxOut (..), UTxO (..) )
+import Control.Monad.Trans.State
+    ( State, evalState, get, put )
+import Data.List
+    ( splitAt )
+import Data.Maybe
+    ( mapMaybe )
+import Data.Word
+    ( Word8 )
+
+import qualified Data.Map.Strict as Map
+
+selectCoinsForMigration
```

Comment:
The body of the function is well explained, but could this top-level function also have a doc string?
Like saying that it produces a series of transactions (~ coin selections) which each have only slightly less outputs than inputs - and this is done so that the the new wallet will have plenty of utxo ready to spend.

Reply:
Good idea! Done :+1: